### PR TITLE
Add ability to send graphs via email after generating them

### PR DESCRIPTION
**Pull request by ascl00**
Simple change to optionally send an email once the graphs are generated.

### CODE REVIEW

**Comment by seren** [comment]
Thanks for the PR. I have a couple questions and comments which I'll leave in-line with the code.

File: rrd-graph.sh [hunk 1]
```diff
@@ -23,7 +23,7 @@
 # Script variables
 ######################################
 # These use external environment variables if set, otherwise use arbitrary default values
-MAXGRAPHTEMP=${MAXGRAPHTEMP:-70}
+MAXGRAPHTEMP=${MAXGRAPHTEMP:-50}
```
Comment (seren) [inline rrd-graph.sh]: Why this change? I've seen drives exceed 50 often (not a great thing I realize).

File: rrd-graph.sh [hunk 2]
```diff
@@ -251,12 +260,58 @@ func_graph_drives_separately () {
...
+    graphs+=( $outputfilename )
     guidrule="HRULE:${SAFETEMPMIN}#0000FF:Min-safe-temp-mechanical:dashes HRULE:${SAFETEMPMAX}#FF0000:Max-safe-temp-mechanical:dashes"
     title="Temperature: Drive ${drdev}, ${interval} minute interval"
     write_graph_to_disk
   done
 }
 
+# Send generated graphs via email
+func_send_email () {
+    local logfile=/tmp/$$.email
```
Comment (seren) [inline rrd-graph.sh]: Could you change this to `local logfile=$(mktemp)`?

File: rrd-graph.sh [hunk 3]
```diff
@@ -265,5 +320,6 @@ func_graph_cpus_together
 func_graph_drives_together
 # func_graph_cpus_separately
 # func_graph_drives_separately
-
-
+if [ ! -z "${email}" ]; then
```
Comment (seren) [inline rrd-graph.sh]: Just curious why `! -z` instead of `-n`?

File: rrd.sh [hunk 4]
```diff
@@ -184,17 +184,17 @@ else
   fi
 fi
 
+tempfile=/tmp/$$.temp.rrd.output
```
Comment (seren) [inline rrd.sh:189]: Could we use `mktemp` here as well?

File: rrd-graph.sh [hunk 5]
```diff
@@ -1,4 +1,4 @@
-#!/usr/local/bin/bash
+#!/usr/bin/env bash
```
Comment (seren) [inline rrd-graph.sh:1]: Good idea. Much more robust.

**Comment by ascl00** [comment]
Sure, ask away.

**Review by seren — CHANGES_REQUESTED** [review]

File: rrd-graph.sh (hunk 1 above)
Comment (ascl00) [reply rrd-graph.sh]: Sorry that wasn't intended to be part of the PR, was supposed to be a local only change (if my drives get over 50 I'll be worried, let alone 70!).

File: rrd-graph.sh (hunk 2 above)
Comment (ascl00) [reply rrd-graph.sh]: Sure, I didn't know there was a mktemp available from the shell. Cool.

Note I kept a prefix on the temp file, because I find it easier when debugging or generally poking around the system if there is some kind of description in the temp file.

File: rrd-graph.sh (hunk 3 above)
Comment (ascl00) [reply rrd-graph.sh]: From: http://www.tldp.org/LDP/abs/html/comparison-ops.html

> The -n test requires that the string be quoted within the test brackets. Using an unquoted string with ! -z, or even just the unquoted string alone within test brackets (see Example 7-6) normally works, however, this is an unsafe practice. Always quote a tested string.

So it's a force of habit to make it slightly harder to introduce a bug. But it isn't a big deal, string is quoted, and inverse logic is harder to read, so I changed it.

File: rrd.sh (hunk 4 above)
Comment (ascl00) [reply rrd.sh:189]: Done
